Feature/lp199 office refactor to text components

AndresIvanMedinaH opened the pull request:
**• Task’s name**
Office -Refactor-to-Text-Components

    **• Task’s link to navigate to jira**
https://devaptivist.atlassian.net/browse/LP-199?atlOrigin=eyJpIjoiNGVmYmNkMzE0YTc3NDg2MmFiYjgzOWI1NTQxNjNjMTUiLCJwIjoiaiJ9

updated-->
![imagen](https://github.com/irv205/droid_leaspert/assets/128720218/b9faa140-5d3f-4ebf-8da8-697b32c4a97b)

AndresIvanMedinaH committed:
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.

+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider
______________________________________________________________
Office ->renamed to -> OfficeView to keep uniformity on naming Views, ViewModels and so

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider
______________________________________________________________
Office ->renamed to -> OfficeView to keep uniformity on naming Views, ViewModels and so

AndresIvanMedinaH committed:
TO-DO
+Verify that the "Information" title is displayed correctly.
+Verify that the address of the office is displayed correctly.
+Verify that the office hours are displayed correctly.
+Verify that the "Call" button opens the phone app with the office phone number.
+Verify that the "Mail" button opens the email app with the office email.

Solution proposition:
based on SOLID:
Use an Interface to:
Grant permission to whoever implements this interface
On Implementation:
Generate the logic that deal with the permission asking based on the different kind of permissions
this implementation requieres the Context where it is need it or being called

use an Interface:
IintentActions that deals with the different permissions needed by intents along the app

On Implementation:
create the necessary methods to process the launching of the intents and receives the inection of the
permissionProvider
______________________________________________________________
Office ->renamed to -> OfficeView to keep uniformity on naming Views, ViewModels and so

AndresIvanMedinaH committed:
TO-DO
+Validations
+Office refactoring of text components to LPText components

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
TO-DO
+Validations
+Office refactoring of text components to LPText components

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH commented:
in this pr you might find the refactoring of the text components, replacing by LPText components

carlosph816 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -0,0 +1,101 @@
+        viewModel.getOfficeInformation()
+    }
+
+    val OfficeData = remember { viewModel.officeInfo }
```
I think you don't need a remember here cause it's already a mutableStateOf and it lives inside of a view model

AndresIvanMedinaH replied:
Fixing it

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Merge remote-tracking branch 'Irving/feature/LP162-Office-viewModel' into feature/LP162-Office-viewModel
# Conflicts:
#	feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt

AndresIvanMedinaH committed:
Merge branch 'develop' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Files modified:
-OfficeModel
-OfficeView
-OfficeViewModel

Files removed
-ContactTelephone
-Email

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Files modified:
-OfficeModel  renamed to Office  moved to Domain
-OfficeView
-OfficeViewModel
-Strings.xml added fields
graddle.build

Files removed
-ContactTelephone
-Email

AndresIvanMedinaH commented:
in this pr you might find the refactoring of the text components, replacing by LPText components:

Files modified :
OfficeView
OfficeViewModel
OfficeModel

Files removed:
Domain
  |-Java
      |-com
            |-itenum
                  |-domain
                        ContactTelephone
                        Email

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/data/OfficeModel.kt:
```diff
@@ -1,8 +1,5 @@
-
 /**
  * This class is for the data for each item in the list.
  * @param address the address of the busines
```
in param address change busines to business

AndresIvanMedinaH replied:
changed the typo
![imagen](https://github.com/irv205/droid_leaspert/assets/128720218/f773d2ab-6fa6-403c-8d5d-625751f9893b)

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/data/OfficeModel.kt:
```diff
@@ -14,6 +11,6 @@ import com.itenum.domain.Email
 data class OfficeModel(
```
Rename OfficeModel to OfficeDTO 

Note: when a model is in data layer is named as DTO, when is in domain Layer it should named as Office or OfficeDomain

AndresIvanMedinaH replied:
already renamed to Office and moved into domain
![imagen](https://github.com/irv205/droid_leaspert/assets/128720218/f019f30f-766d-454a-aaf6-1ee082104655)

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -35,7 +35,7 @@ fun OfficeView() {
-    val officeData =  viewModel.officeInfo
+    val officeData = viewModel.officeInfo
 
     officeData?.let {
```
Remember to make a reformatting code for easy reading.  Use **ctlr + L** to reformat code

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -71,13 +71,13 @@ fun OfficeUI(
+                    weight = FontWeight.Bold,
                     onClick = {})
                 LPBodyMedium(modifier = Modifier.padding(top = sizes.extraSize10),
                     label = officeData.address,
```
missing label **Address** in the UI
![image](https://github.com/irv205/droid_leaspert/assets/33376944/7402403d-b77b-4ca0-bc01-0bba598f7df0)

AndresIvanMedinaH replied:
word Address is there, but the frame is hiding that part

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -71,13 +71,13 @@ fun OfficeUI(
+                    weight = FontWeight.Normal,
                     onClick = {})
                 LPBodyMedium(modifier = Modifier.padding(top = sizes.extraSize10),
                     label = stringResource(id = R.string.LPHours) + officeData.schedule,
```
Instead of using a **+** after the **stringResource(id = R.string.LPHours)** you can add it as string resource parametter, check this for a better understanding.
![image](https://github.com/irv205/droid_leaspert/assets/33376944/5ad09c71-78b6-46fd-b9ff-a16d60bcab40)

Add it please, the same for Address label in stringResource =)

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Files modified:
-OfficeModel  renamed to Office  moved to Domain
-OfficeView
-OfficeViewModel
-Strings.xml added fields
graddle.build

Files removed
-ContactTelephone
-Email

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

Lupillo17 commented on feature/office/build.gradle:
```diff
@@ -35,6 +35,7 @@ android {
 
 dependencies {
     implementation libs.hiltAndroid
```
Remove **hiltAndroid** implementation

Note: **libs.bundles.hilt** contains **hiltAndroid**

AndresIvanMedinaH replied:
removed
![imagen](https://github.com/irv205/droid_leaspert/assets/128720218/6b1e6fed-41a6-49bc-8b85-5c8fa0f0f705)

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -28,20 +30,45 @@ import com.iteneum.office.data.OfficeModel
+        }
+
+        is OfficeStateResponse.Success -> {
+            val OfficeData = viewModel.officeInfo
```
rename **OfficeData**, first letter should be in lower case, variables should be named in lower case

AndresIvanMedinaH replied:
lowered 
![imagen](https://github.com/irv205/droid_leaspert/assets/128720218/76407dfa-eff7-4f1f-9c93-4edf893b6208)

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -50,55 +77,51 @@ fun OfficeView() {
  * @param onEmailButtonClicked launches the Email intent
  * @author Yaritza Moreno
  * Modified by:
```
Remove line 79 **Modified by:** already exists **@modififiedBy** below next line

AndresIvanMedinaH replied:
got it

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -50,55 +77,51 @@ fun OfficeView() {
+            weight = FontWeight.Bold,
+            onClick = {})
+        LPBodyMedium(modifier = Modifier.padding(top = sizes.extraSize10),
+            label = officeData.address,
```
in named param **label** add stringResource() and try add **officeData.address** as stringResource() parameter.

Here is an example
![image](https://github.com/irv205/droid_leaspert/assets/33376944/6978e00b-ed4b-432d-bd58-08055ef47d9d)

AndresIvanMedinaH replied:
added:
val address = stringResource(id = R.string.LPAddress,officeData.address)
val schedule = stringResource(id = R.string.LPHours,officeData.schedule)

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeView.kt:
```diff
@@ -50,55 +77,51 @@ fun OfficeView() {
+            weight = FontWeight.Normal,
+            onClick = {})
+        LPBodyMedium(modifier = Modifier.padding(top = sizes.extraSize10),
+            label = stringResource(id = R.string.LPHours) + officeData.schedule,
```
use stringResource() and pass **officeData.schedule** as parameter

AndresIvanMedinaH replied:
Hours are supose to be a whole string?  or just the hours like in:

Hours: [  9am to 6pm ]

AndresIvanMedinaH replied:
changed all the string into the object response

Lupillo17 commented on feature/office/src/main/res/values/strings.xml:
```diff
@@ -7,4 +7,6 @@
     <string name="LPAddress">Address: 4950 Gaidrew, Alpharetta, GA, 30022</string>
     <string name="LPphoneServiceContact">+1 452 123 4567</string>
     <string name="LPcontactEmail">contact@leaspert.com</string>
+    <string name="Error">Error\n</string>
```
remove **\n** from both Error and Loading

AndresIvanMedinaH replied:
erased!

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Files modified:
-OfficeModel  renamed to Office  moved to Domain
-OfficeView
-OfficeViewModel
-Strings.xml added fields
graddle.build

Files removed
-ContactTelephone
-Email

AndresIvanMedinaH committed:
TO-DO
+view model implementation

Solution proposition:
OfficeView--> Text() Not refactored!!! review --> feature/LP199
OfficeVideModel --> refactored without permissions implementation
______________________________________________________________
Files modified:
-OfficeModel  renamed to Office  moved to Domain
-OfficeView
-OfficeViewModel
-Strings.xml added fields
graddle.build

Files removed
-ContactTelephone
-Email

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

Lupillo17 commented on feature/office/src/main/java/com/iteneum/office/presentation/ui/OfficeViewModel.kt:
```diff
@@ -65,8 +66,8 @@ class OfficeViewModel() : ViewModel() {
-        "4950 Gaidrew, Alpharetta, GA, 30022 ",
-        "9AM-6PM",
+        "4950 Gaidrew, Alpharetta, GA, 30022",
+        "Hours: 9AM-6PM",
```
Replace **Hours: 9AM-6PM** to **9AM-6PM** without **Hours**
Aditionally add named params for **address** and **schedule**

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components

AndresIvanMedinaH committed:
Merge remote-tracking branch 'Irving/feature/LP199-Office-Refactor-to-Text-Components' into feature/LP199-Office-Refactor-to-Text-Components
